jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/WebSocketConnection.java

```diff
@@ -200,10 +200,39 @@ public void setUseOutputDirectByteBuffers(boolean useOutputDirectByteBuffers)
     }
 
     /**
-     * Physical connection disconnect.
-     * <p>
-     * Not related to WebSocket close handshake.
+     * Extra bytes from the initial HTTP upgrade that need to
+     * be processed by the websocket parser before starting
+     * to read bytes from the connection
+     *
+     * @param initialBuffer the bytes of extra content encountered during upgrade
      */
+    protected void setInitialBuffer(ByteBuffer initialBuffer)
+    {
+        if (LOG.isDebugEnabled())
+            LOG.debug("Set initial buffer - {}", BufferUtil.toDetailString(initialBuffer));
+        try (AutoLock l = lock.lock())
+        {
+            networkBuffer = newNetworkBuffer(initialBuffer.remaining());
+        }
+        ByteBuffer buffer = networkBuffer.getByteBuffer();
+        BufferUtil.clearToFill(buffer);
+        BufferUtil.put(initialBuffer, buffer);
+        BufferUtil.flipToFlush(buffer, 0);
+    }
+
+    @Override
+    public void onOpen()
+    {
+        if (LOG.isDebugEnabled())
+            LOG.debug("onOpen() {}", this);
+
+        // Open Session
+        super.onOpen();
+        coreSession.onOpen();
+        if (moreDemand())
+            fillAndParse();
```

Comment:
Ah this is the fix.  Hard to see as the methods were moved for some reason???

Reply:
Actually, for better history, can you move these methods back to where they were?

Reply:
I'll move it back.

Although the ordering of the methods in this class is pretty unusual.
I would prefer to see `onOpen` and `onClose` grouped together, instead of `onClose` at the very top and `onOpen` at the very bottom.